include/swift/SIL/SILDebugScope.h

```diff
@@ -36,18 +37,20 @@ class SILInstruction;
 /// debug info. In contrast to LLVM IR, SILDebugScope also holds all
 /// the inlining information. In LLVM IR the inline info is part of
 /// DILocation.
-class SILDebugScope : public SILAllocated<SILDebugScope> {
+class SILDebugScope final
+    : public SILAllocated<SILDebugScope>,
+      private llvm::TrailingObjects<SILDebugScope, const SILDebugScope *> {
+  friend TrailingObjects;
+
+public:
+  using ParentType = PointerUnion<const SILDebugScope *, SILFunction *>;
```

Comment:
Out of curiosity: why is this preferred over a typedef? Is that just a style thing?

Reply:
It's much better for function and array types because the name isn't embedded in the middle, and then I'd rather be consistent across everything. But yeah, in this case it doesn't really matter.

(Also `using` can handle templates, but that doesn't matter here either.)